### PR TITLE
builtins: implement ST_RemoveRepeatedPoints

### LINKED ISSUE #49017: geo/geomfn: implement ST_RemoveRepeatedPoints({geometry,float8})
Implement `ST_RemoveRepeatedPoints` on arguments {geometry,float8}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_RemoveRepeatedPoints.html).

*Observers: Please react to this issue if you need this functionality.*

For Geometry builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geomfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geomfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
  * When using GEOS, you can reference the [C API](https://github.com/libgeos/geos/blob/master/capi/geos_c.h.in) for which functions are available. Unfortunately, Windows is not currently supported when using GEOS.
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end (your exhaustive unit tests go the above mentioned packages!).
* Ensure the documentation is regenerated by calling `make buildshort`. You can also play with it by calling `./cockroach demo --empty` afterwards.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48552 for an example PR.

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2020-08-30T17:39:33Z. Changes to titles, body and labels may be overwritten.</sub>

### PR DESCRIPTION
**Pull request by erikgrinaker**
Release justification: low risk, high benefit changes to existing functionality

Release note (sql change): Implement the geometry builtin
`ST_RemoveRepeatedPoints`.

Closes #49017.

### CODE REVIEW

**Comment by otan** [comment]
This is interesting. I had to do something similar in https://github.com/cockroachdb/cockroach/pull/53970 but with the caveat that it seems to always keep the first coordinates:

```
otan=# select st_astext(st_removerepeatedpoints('linestring(1 2, 1 2)'::geometry));
      st_astext
---------------------
 LINESTRING(1 2,1 2)
(1 row)

otan=# select st_astext(st_removerepeatedpoints('linestring(1 2, 1 2, 3 4)'::geometry));
      st_astext
---------------------
 LINESTRING(1 2,3 4)
(1 row)

otan=# select st_astext(st_removerepeatedpoints('linestring(1 2, 1 2, 1 2, 3 4)'::geometry));
      st_astext
---------------------
 LINESTRING(1 2,3 4)
(1 row)

otan=# select st_astext(st_removerepeatedpoints('linestring(1 2, 1 2, 1 2)'::geometry));
      st_astext
---------------------
 LINESTRING(1 2,1 2)
(1 row)
```

can you verify that if it's all the same coordinates behaviors match too?

File: pkg/geo/geomfn/remove_repeated_points_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		expected  string
+	}{
+		{"MULTIPOINT (1 1, 2 2, 3 3, 4 4, 5 5)", 1.4, "MULTIPOINT (1 1, 2 2, 3 3, 4 4, 5 5)"},
+		{"MULTIPOINT (1 1, 2 2, 3 3, 4 4, 5 5)", 1.5, "MULTIPOINT (1 1, 3 3, 5 5)"},
+		{"MULTIPOINT (1 1, 2 2, 3 3, 4 4, 5 5)", 3, "MULTIPOINT (1 1, 4 4)"},
+		{"MULTIPOINT (1 1, 2 2, 3 3, 4 4, 5 5)", 6, "MULTIPOINT (1 1)"},
+	}
+
+	for _, tc := range testCasesTolerance {
+		t.Run(tc.wkt, func(t *testing.T) {
```
Comment (otan) [inline pkg/geo/geomfn/remove_repeated_points_test.go:89]: mind appending the tolerance to the tc description?

**Comment by erikgrinaker** [comment]
> st_removerepeatedpoints seems to always keep the first two coordinates

Good catch, thanks - these cases errored, it needs to enforce all type invariants (e.g. that linestrings have at least 2 coords, and that polygon linearrings have at least 4 coords and are closed). I've fixed the cases I could think of and added tests for them.

**Review by otan — APPROVED** [review]
neat, thanks!

**Pull request merged**